packages/vite/src/node/plugins/define.ts

```diff
@@ -123,17 +124,11 @@ export function definePlugin(config: ResolvedConfig): Plugin {
         return null
       }
 
-      if (ssr && !isBuild) {
```

Comment:
I don't think it's feasible to use `stripLiteral` in this code path, so I removed this optimization. SSR dev now uses `MagicString` to perform the string replacements to take advantage of this fix. Let me know what you think.